numpy/distutils/command/config.py

```diff
@@ -96,8 +94,8 @@ def _wrap_method(self, mth, lang, args):
             self.compiler = self.fcompiler
         try:
             ret = mth(*((self,)+args))
-        except (DistutilsExecError, CompileError):
-            str(get_exception())
+        except (DistutilsExecError, CompileError) as e:
+            str(e)
```

Comment:
Does this actually do anything/

Reply:
I mean, in principle it could if someone declares:
```python
class WeirdException(CompileError):
    def __str__(self):
       print("Yes this does something")
       return CompilerError.__str__(self)
```

I was trying to keep the changes in this patch strictly to removing workarounds `except E as e:`, rather than evaluating whether `e` was used sensibly in the first place.

Reply:
Ha, it was intended to fix an unused variable warning in #12448. 
```
-            msg = str(get_exception())
￼+           str(get_exception())
```

Reply:
My guess is that the fixer thought there might be a side effect in the function call.

Reply:
Thanks for the detective work - I think I'd still rather leave that to a follow-up PR - searching for `str(e)` probably finds a large amount of stuff that can be better written with exception chaining.

Reply:
Flake8 doesn't like that file :) But it doesn't complain about unused variables. I think we can improve a lot of messages with `f-strings`. We could do that now that Python 3.5 has been dropped, but probably better to wait for the next release.